ui/vendor/_directional.scss

```diff
@@ -0,0 +1,63 @@
+// directional-scss | Author: Tyson Matanich (http://matanich.com), 2013 | License: MIT
+$dir: ltr !default;
+
+// Default $dir if not valid
+@if $dir != ltr and $dir != rtl {
+    $dir: ltr;
+}
+
+@function if-ltr($if, $else: null) {
+  @if $dir != rtl {
+    @return $if;
+  }
+  @else {
+    @return $else;
+  }
+}
+
+@function if-rtl($if, $else: null) {
+  @return if-ltr($else, $if);
+}
+
+$left: if-ltr(left, right);
+$right: if-ltr(right, left);
```

Comment:
I've been reworking the code so that `left` stays left. As I mentioned about, where this gets tricky is when a component (e.g. activity timeline) uses `slds-m-left--large`. Given what's been said about text alignment and `left` always being `left`, should we also apply the same logic for padding/margins? I.e. people would need to change their DOM to use `slds-m-start--large` instead and therefore get access to rtl features?

The big downside I see to this is that it means the docs have been pushing people in a non-rtl compatible direction and everyone would need to change their dom to get `rtl` for free. Whereas if we do allow `slds-m-left--large` to actually be `margin-right` when embedded in a `[dir=rtl]` parent means that all components can become `rtl` compatible out of the box. So when Lightning Experience finally does support RTL, people's custom Lightning components won't automatically be rtl compatible. Maybe that's a good thing? Because chances are most people don't initially write them with rtl support in mind.

What do others think about this? Should `left` stay meaning `left` and we introduce `start`/`end` versions of everything that people migrate to for rtl support?